microprofile-metrics/src/test/java/org/jboss/eap/qe/microprofile/metrics/integration/ft/FTMetricTest.java

```diff
@@ -0,0 +1,207 @@
+    }
+
+    /**
+     * @tpTestDetails Multi-component customer scenario to verify correct behavior when custom counter metric is used.
+     *                The counter increment value depends on CDI bean and MP Config property values. The increment for the
+     *                counter
+     *                is provided by a fail-safe service. If one provider fails (controlled by MP Config property)
+     *                the fail-safe service fallbacks to another provider.
+     * @tpPassCrit Counter metric is incremented by configured value change dynamically
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    public void incrementFailSafeProviderTest() throws IOException {
+        setMPConfig(false, 1, 3);
+        performRequest();
+        performRequest();
+        testCustomMetricForValue(2);
+        testInvocationsMetric(2);
+        setMPConfig(true, 2, 4);
+        performRequest();
+        performRequest();
+        performRequest();
+        testCustomMetricForValue(14);
+        testInvocationsMetric(5);
+        testFallbackMetric(3);
+        setMPConfig(false, 10, 6);
+        performRequest();
+        testCustomMetricForValue(24);
+        testInvocationsMetric(6);
+        testFallbackMetric(3);
+    }
+
+    private void setMPConfig(boolean providerCorrupted, int increment, int failSafeIncrement) throws IOException {
+        String content = String.format("%s=%s\n%s=%s\n%s=%s",
+                FAULT_CTL_CONFIG_PROPERTY, providerCorrupted,
+                INCREMENT_CONFIG_PROPERTY, increment,
+                FAILSAFE_INCREMENT_CONFIG_PROPERTY, failSafeIncrement);
+        FileUtils.writeStringToFile(SetupTask.propertyFile, content);
+    }
+
+    private void testCustomMetricForValue(int value) {
+        testAppCounterMetric("ft-custom-metric", value);
+    }
+
+    private void testInvocationsMetric(int value) {
+        testAppCounterMetric(
+                "ft." + FTCustomCounterIncrementProviderService.class.getName() + ".getIncrement.invocations.total", value);
+    }
+
+    private void testFallbackMetric(int value) {
+        testAppCounterMetric(
+                "ft." + FTCustomCounterIncrementProviderService.class.getName() + ".getIncrement.fallback.calls.total", value);
+    }
+
+    private void testAppCounterMetric(String name, int value) {
+        given()
+                .baseUri("http://" + managementClient.getMgmtAddress() + ":" + managementClient.getMgmtPort() + "/metrics")
+                .accept(ContentType.JSON)
+                .get()
+                .then()
+                .body("application", hasKey(name),
+                        "application.'" + name + "'", equalTo(value));
+
+    }
+
+    private void performRequest() {
+        get(deploymentUrl.toString()).then()
+                .statusCode(200)
+                .body(equalTo("Hello from custom metric fault-tolerant service!"));
+    }
+
+    static class SetupTask implements ServerSetupTask {
```

Comment:
Can you please document this Setuptask? Its `setup` method seems more complex.

Reply:
Fixing...